### ISSUE TITLE
Makefiles generate a lot of targets that users typically don't invoke directly

### ISSUE THREAD

**Issue by EricHripko** [microsoft/vscode-makefile-tools#60]
# Overview
My project uses a `Makefile` that specifies a lot of system/library headers as target dependencies (see sample output from `targets.log`):
```
exmpllogic_bestdataproviderdb.linux.o: ... /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/atomic_base.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/atomic_lockfree_defines.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/basic_ios.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/basic_ios.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/basic_string.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/basic_string.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/char_traits.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/codecvt.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/concept_check.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/cpp_type_traits.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/cxxabi_forced.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/cxxabi_init_exception.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/enable_special_members.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/exception.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/exception_defines.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/exception_ptr.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/functexcept.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/functional_hash.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/hash_bytes.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/hashtable.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/hashtable_policy.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/invoke.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/ios_base.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/istream.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_classes.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_classes.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_conv.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_facets.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_facets.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_facets_nonio.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/locale_facets_nonio.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/localefwd.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/memoryfwd.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/move.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/nested_exception.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/node_handle.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/ostream.tcc /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/ostream_insert.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/postypes.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/predefined_ops.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/ptr_traits.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/quoted_string.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/range_access.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/refwrap.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/shared_ptr.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/shared_ptr_atomic.h /opt/rh/devtoolset-7/root/usr/include/c++/7/bits/shared_ptr_base.h ...
#  Implicit rule search has not been done.
#  Modification time never checked.
#  File has not been updated.
```
This leads to the extension creating a huge number of targets - making it difficult to find the targets that users are expected to use.

# Expected Behaviour
Ideally, I'd like to only see a small subset of targets that are commonly used, for example: clean/build/install/all. In my case, this could possibly be generated from `PHONY` targets, but I unsure whether this applies to all `Makefile` projects.

# Actual Behaviour
Every system/library/app header and object is listed as a target.

**Comment by andreeis** [comment]
@EricHripko, we released Makefile Tools 0.1.2 which contains the .PHONY targets filter that you suggested.
https://github.com/microsoft/vscode-makefile-tools/releases/tag/v0.1.2.
Let us know how 0.1.2 works for you. Thank you very much for dogfooding this pre-release.

**Comment by EricHripko** [comment]
This has worked brilliantly! `"makefile.phonyOnlyTargets": true` reduced _1350_ targets to just _3_ high-level ones 🎉 Thank you for actioning the feedback and investing into this extension 🙂
